picard/tagger.py

```diff
@@ -1601,16 +1609,67 @@ def setup_dbus():
         pass
 
 
-def setup_translator(tagger):
-    """Setup Qt default translations"""
-    translator = QtCore.QTranslator(tagger)
-    locale = QtCore.QLocale()
-    translation_path = QtCore.QLibraryInfo.path(QtCore.QLibraryInfo.LibraryPath.TranslationsPath)
-    log.debug("Looking for Qt locale %s in %s", locale.name(), translation_path)
-    if translator.load(locale, 'qtbase_', directory=translation_path):
-        tagger.installTranslator(translator)
-    else:
-        log.debug("Qt locale %s not available", locale.name())
+# Translator priority constants (higher = installed later = searched first)
+TRANSLATOR_PRIORITY_QT_BASE = 100  # Qt base translations (highest priority)
+TRANSLATOR_PRIORITY_PLUGIN = 0  # Plugin translations (lower priority)
+
+
+@dataclass(init=True, order=True, kw_only=True)
+class Translator:
+    sort_index: int = TRANSLATOR_PRIORITY_QT_BASE
+    installed: bool = field(default=False, compare=False)
+    instance: QtCore.QTranslator = field(default=None, compare=False)
+
+
+class Translators:
+    def __init__(self, tagger):
+        self.tagger = tagger
+        self.tagger.translations_updated.connect(self.reinstall)
+        self._translators = []
+        self.add_default_translators()
+
+    def add_default_translators(self):
+        translator = QtCore.QTranslator(self.tagger)
+        locale = QtCore.QLocale()
+        translation_path = QtCore.QLibraryInfo.path(QtCore.QLibraryInfo.LibraryPath.TranslationsPath)
+        log.debug("Looking for Qt locale %s in %s", locale.name(), translation_path)
+        if translator.load(locale, 'qtbase_', directory=translation_path):
+            t = Translator(sort_index=TRANSLATOR_PRIORITY_QT_BASE, instance=translator)
+            self._translators.append(t)
+        else:
+            log.debug("Qt locale %s not available", locale.name())
+
+    def add_translator(self, translator):
+        t = Translator(sort_index=TRANSLATOR_PRIORITY_PLUGIN, instance=translator)
+        self._translators.append(t)
+
+    def remove_translator(self, translator):
+        for t in self._translators[:]:
+            if t.instance == translator:
+                if t.installed:
+                    log.debug("Remove translator: %r", t)
+                    self.tagger.removeTranslator(t.instance)
+                self._translators.remove(t)
+                break
+
+    def reinstall(self):
+        log.debug("Reinstall translators")
+
+        # Translations are searched for in the reverse order in which they were installed,
+        # so the most recently installed translation file is searched for translations first
+        # and the earliest translation file is searched last.
+        # The search stops as soon as a translation containing a matching string is found.
+
+        # First, remove installed translators
+        for t in self._translators:
+            if t.installed:
+                self.tagger.removeTranslator(t.instance)
+                t.installed = False
```

Comment:
Alternatively to the suggested changed status we could check in this loop if any translator `t.installed == False`. If not, there is actually nothing to add, so we can skip updating the translators.

Main point I want to avoid this is that q triggers the signal that translations have changed, and we can avoid work in the UI if not needed.